drivers/usb/device/usb_dc_stm32.c

```diff
@@ -444,6 +445,9 @@ static int usb_dc_stm32_init(void)
 	}
 #endif /* USB */
 
+	/* Disable soft disconnect */
+	USB_DEVICE_REG->DCTL &= ~USB_OTG_DCTL_SDIS;
```

Comment:
Isn't it something do-able via the HAL? I'm a bit reluctant writing dev register directly here, is it compatible with all STM USB controllers?

Reply:
Okay previous comment was for the detach, but for the attach we want:
```c
HAL_StatusTypeDef HAL_PCD_DevConnect(PCD_HandleTypeDef *hpcd)
{
  USB_OTG_GlobalTypeDef *USBx = hpcd->Instance;

  __HAL_LOCK(hpcd);

  if ((hpcd->Init.battery_charging_enable == 1U) &&
      (hpcd->Init.phy_itface != USB_OTG_ULPI_PHY))
  {
    /* Enable USB Transceiver */
    USBx->GCCFG |= USB_OTG_GCCFG_PWRDWN;
  }
  (void)USB_DevConnect(hpcd->Instance);
  __HAL_UNLOCK(hpcd);

  return HAL_OK;
}

/* or the lower one */
HAL_StatusTypeDef  USB_DevConnect(USB_OTG_GlobalTypeDef *USBx)
{
  uint32_t USBx_BASE = (uint32_t)USBx;

  /* In case phy is stopped, ensure to ungate and restore the phy CLK */
  USBx_PCGCCTL &= ~(USB_OTG_PCGCCTL_STOPCLK | USB_OTG_PCGCCTL_GATECLK);

  USBx_DEVICE->DCTL &= ~USB_OTG_DCTL_SDIS;

  return HAL_OK;
}
```
I'm going to assume you agree, and push the changes, if someone disagrees, or sees that it may have consequences can always comment.